python_modules/dagster/dagster/_core/remote_representation/external.py

```diff
@@ -466,6 +413,18 @@ def _sensor_mappings(
                 try:
                     keys = sensor.asset_selection.resolve(self.asset_graph)
                     for key in keys:
+                        # only count an asset as targeted by an automation condition sensor if it
+                        # has an automation condition
+                        if sensor.sensor_type in (
+                            SensorType.AUTO_MATERIALIZE,
+                            SensorType.AUTOMATION,
+                        ):
+                            node_snap = self.get_asset_node_snap(key)
+                            if not node_snap or not (
+                                node_snap.automation_condition
+                                or node_snap.automation_condition_snapshot
+                            ):
+                                continue
```

Comment:
i'm a little unclear how this part relates to the PR - does the fact that these are now sometimes defined in user code now cause issues here?

Reply:
I'll be honest, I don't know why i lumped this change in (it's been awhile), but the basic idea is that we use the "targeted by" to render stuff in the UI and it feels a little silly if we link to the automation condition sensor from the asset details page of an asset that doesn't have a condition on it.